sympy/polys/polytools.py

```diff
@@ -1000,6 +1006,32 @@ def as_expr(f, *gens):
 
         return basic_from_dict(f.rep.to_sympy_dict(), *gens)
 
+    def as_poly(self, *gens, **args):
```

Comment:
I don't think this would be much needed for `Poly` objects. Maybe the code could just be something like

     return self.as_expr().as_poly(*gens, **args)

Reply:
Should it be possible to pass a `Poly` to factor like: `factor(Poly(x**2+x))` or should that use `as_expr`? I think it should use `as_expr` so that factor only needs to expect `Expr`.

Reply:
> so that factor only needs to expect Expr

That is what I think most users would expect. Besides, there is a `Poly.factor_list` method.

Reply:
`Poly.as_poly` is used heavily in risch so it would be good to have some method for this even if it isn't called as_poly.

> ```
>  return self.as_expr().as_poly(*gens, **args)
> ```

The situation in which this fails is if the poly is constant and no generators are provided:
```julia
In [2]: gens = []                                                                                                                              

In [3]: Poly(1, x).as_expr().as_poly(*gens)                                                                                                    
---------------------------------------------------------------------------
GeneratorsNeeded 
```

Reply:
> Poly.as_poly is used heavily in risch

Hmm... I didn't remember that. Maybe something like this copied code is then needed, though I wonder if those calls could be replaced by something else in risch.

Reply:
I don't know risch well enough to mess with it so I would only replace it with something that does exactly the same thing which doesn't really seem worthwhile.

Maybe @asmeurer has a better idea...

Reply:
I think this code and the calls in risch can be left as is for now. (It looks like their purpose is to change gens and/or domain.) But maybe the docstring examples could be replaced by one with `Poly` type `self`.